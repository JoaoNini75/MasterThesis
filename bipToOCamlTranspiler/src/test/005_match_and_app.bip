let assert1 () = begin
	assert (1 <> 3)
end
(*@ ensures true *)

let assert2 (x: int) : int = begin
	assert (x + 4 == 5);
	x - 1
end
(*@ ensures true *)

let app_fun (x : int) = begin
	x
end
(*@ ensures true *)

let app_fun2 () = begin
	()
end
(*@ ensures true *)

let app_fun3 (x, y) : int = begin
	x + y
end
(*@ ensures true *)

let app_mid_test (x : int, b : bool) = begin
	app_fun2 ();
	2 + 2
end
(*@ ensures true *)

let app_let_test (x : int, b : bool) = begin
	let app_res = app_fun (1) in
	app_res
end
(*@ ensures true *)

let app_end_test (x : int, b : bool) = begin
	app_fun (1)
end
(*@ ensures true *) 

let apptestfinal () : int = begin
	let a = ref 4 in
	app_fun3 (1 * !a, 2 * !a)
end
(*@ ensures true *)

let match1 (x : int, y : int) : string = begin
  match x with 
	| 0 -> "zero"
	| y -> "y"
end
(*@ ensures true *) 

let match2 (x : int) = begin
	let res = 
		match x with 
		| 0 -> 10
		| 1 -> 20
		| _ -> 0
	in
	assert (res >= 0);
	res
end
(*@ ensures true *) 
