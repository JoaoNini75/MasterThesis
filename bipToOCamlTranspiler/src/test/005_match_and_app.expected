let app_fun (x : int) =
  x
(*@ ensures true *)

let app_fun2 () =
  ()
(*@ ensures true *)

let app_fun3 (x) (y) : int =
  (x + y)
(*@ ensures true *)

let app_mid_test (x : int) (b : bool) =
  app_fun2 ();
  (2 + 2)
(*@ ensures true *)

let app_let_test (x : int) (b : bool) =
  let app_res = app_fun (1) in
  app_res
(*@ ensures true *)

let app_end_test (x : int) (b : bool) =
  app_fun (1)
(*@ ensures true *)

let apptestfinal () : int =
  let a = ref (4) in
  app_fun3 ((1 * !a)) ((2 * !a))
(*@ ensures true *)

let match1 (x : int) (y : int) : string =
  match x with
  | 0 -> "zero"
  | y -> "y"
(*@ ensures true *)

let match2 (x : int) =
  let res = (
    match x with
    | 0 -> 10
    | 1 -> 20
    | _ -> 0
  ) in
  assert ((res >= 0));
  res
(*@ ensures true *)


